tests/integration/istio-injection/istioinjection/fixture.go

```diff
@@ -0,0 +1,243 @@
+		if err != nil {
+			log.Errorf("Cannot create in-cluster config: %v", err)
+			panic(err)
+		}
+		return cfg
+	}
+
+	cfg, err := clientcmd.BuildConfigFromFlags("", clientcmd.RecommendedHomeFile)
+	if err != nil {
+		log.Errorf("Cannot read kubeconfig: %s", err)
+		panic(err)
+	}
+	return cfg
+}
+
+func (testSuite *TestSuite) initK8sClient() {
+	kubeConfig := loadKubeConfigOrDie()
+	testSuite.k8sClient = kubernetes.NewForConfigOrDie(kubeConfig)
+}
+
+func (testSuite *TestSuite) disableInjectionForNamespace(disable bool) {
+	var err error
+	var ns *corev1.Namespace
+	err = retry.Do(func() error {
+		ns, err = testSuite.k8sClient.CoreV1().Namespaces().Get(testSuite.namespace, metav1.GetOptions{})
+		return err
+	},
+		retry.Attempts(5),
+		retry.Delay(3*time.Second))
+
+	if err != nil {
+		log.Errorf("Cannot get namespace '%s': %v", testSuite.namespace, err)
+		panic(err)
+	}
+
+	if disable {
+		ns.ObjectMeta.Labels = map[string]string{
+			"istio-injection": "disabled",
+		}
+	} else {
+		ns.ObjectMeta.Labels = map[string]string{}
+	}
+
+	err = retry.Do(func() error {
+		_, err = testSuite.k8sClient.CoreV1().Namespaces().Update(ns)
+		return err
+	},
+		retry.Attempts(5),
+		retry.Delay(3*time.Second))
+
+	if err != nil {
+		log.Errorf("Cannot update namespace '%s': %v", testSuite.namespace, err)
+		panic(err)
+	}
+}
+
+func (testSuite *TestSuite) createDeployment(name string, disableSidecarInjection bool) (*appv1.Deployment, error) {
+	log.Infof("Creating deployment '%s", name)
+	labels := testSuite.labels(name)
+
+	annotations := make(map[string]string)
+	if disableSidecarInjection {
+		annotations["sidecar.istio.io/inject"] = "false"
+	}
+
+	deployment := &appv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: testSuite.namespace,
+			Labels:    labels,
+		},
+		Spec: appv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      labels,
+					Annotations: annotations,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            fmt.Sprintf("cont-%s", name),
+							Image:           HTTPDBSERVICE_IMAGE,
+							ImagePullPolicy: "IfNotPresent",
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "http",
+									ContainerPort: 8017,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	var dep *appv1.Deployment
+	err := retry.Do(func() error {
+		deployment, err := testSuite.k8sClient.AppsV1().Deployments(testSuite.namespace).Create(deployment)
+		dep = deployment
+		return err
+	},
+		retry.Attempts(5),
+		retry.Delay(3*time.Second))
+
+	if err != nil {
+		log.Errorf("Cannot create deployment '%s': %v", deployment, err)
+	}
+
+	return dep, err
+}
+
+func (testSuite *TestSuite) getPods(appLabelValue string) (*v1.PodList, error) {
+	var pods *v1.PodList
+	var err error
+
+	err = retry.Do(func() error {
```

Comment:
You don't need to declare the error, just use `err :=`

Reply:
I cannot do `pods, err :=` inside retry block, therefore I must declare error outside. The `pods` is used outside the retry block.